### DISCUSSION TITLE
preview-key would conflict with vertico-map's keybind

### DISCUSSION THREAD

**Discussion by honmaple** [minad/consult#1085] [Q&A] [answered]
```
(define-key vertico-map [tab] 'vertico-insert)

(consult-customize
 consult-line
 :preview-key "<tab>")
```

When I use `consult-line` search something, then use `tab` to preview, this action would be `vertico-insert` and `preview`, how can I disable this behaviour? I just want to `preview` rather then `vertico-insert` and `preview`

This behaviour maybe useful only binding `up` or `down`. Thanks
https://github.com/minad/consult/wiki#configure-s-ups-down-preview-keys

**Comment by minad** [comment]
You can bind `TAB` to `ignore` in `vertico-map`.

**Reply by honmaple** [reply, marked as answer]
@minad Thank you.

For anyone has same problem,use `preview-key` with `keymap`
```
  (defvar maple/consult-preview-map
    (let ((map (make-sparse-keymap)))
      (define-key map (kbd "<tab>") #'ignore)
      map))

  (consult-customize
   consult-ripgrep
   :keymap maple/consult-preview-map
   :preview-key "<tab>"
   consult-line
   :preview-key '(:debounce 0.2 any))
```

If don't want to add every custom command to consult-customize,such as 
```
  (defun maple/consult-grep (&optional dir initial)
    (interactive "P")
    (consult-ripgrep (or dir default-directory) initial))
```
just use 
```
  (defun maple/consult-grep (&optional dir initial)
    (interactive "P")
    (let ((this-command 'consult-ripgrep))
      (consult-ripgrep (or dir default-directory) initial)))
```
